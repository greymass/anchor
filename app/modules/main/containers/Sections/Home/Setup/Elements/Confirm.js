// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { withTranslation } from 'react-i18next';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import compose from 'lodash/fp/compose';

import { Button, Checkbox, Form, Header, Icon, Message, Segment, Table } from 'semantic-ui-react';

import GlobalButtonElevate from '../../../../../../../shared/containers/Global/Button/Elevate';
import { changeModule } from '../../../../../actions/navigation';
import AccountSetupElementsLedger from './Ledger';
import * as HardwareLedgerActions from '../../../../../../../shared/actions/hardware/ledger';

import AccountSetupElementsLedgerOptionsLedgerAll from './Ledger/Options/LedgerAll';
import AccountSetupElementsLedgerOptionsLedgerRecover from './Ledger/Options/LedgerRecover';
import AccountSetupElementsLedgerOptionsLedgerUse from './Ledger/Options/LedgerUse';

const { ipcRenderer } = require('electron');

class AccountSetupElementsConfirm extends Component<Props> {
  state = {
    ledgerMethod: false,
    ledgerSetup: false,
  }
  onLedgerSetup = (e, { name, checked }) => {
    const { actions, ledger } = this.props;
    this.setState({
      ledgerMethod: false,
      ledgerSetup: checked,
    });
    if (checked && !ledger.listening) {
      actions.ledgerStartListen();
    }
  }
  onLedgerCancel = () => {
    this.setState({
      ledgerMethod: false,
      ledgerSetup: false,
    });
  }
  onLedgerReset = () => {
    this.setState({
      ledgerMethod: false,
      ledgerSetup: true,
    })
  }
  onLedgerComplete = ({method}) => {
    this.setState({
      ledgerMethod: method,
      ledgerSetup: false,
    });
  }
  render() {
    const {
      accountName,
      blockchain,
      loader,
      onBack,
      onClick,
    } = this.props;
    const {
      ledgerSetup,
      ledgerMethod,
    } = this.state;
    const disable = (ledgerSetup && !ledgerMethod);
    return (
      <Segment clearing>
        <Header>
          Confirm account creation
          <Header.Subheader>
            Please confirm the details of your new account.
          </Header.Subheader>
        </Header>
        <AccountSetupElementsLedger
          onCancel={this.onLedgerCancel}
          onComplete={this.onLedgerComplete}
          open={ledgerSetup}
        />
        {loader}
        <Form>
          <Table
            attached={(!ledgerMethod) ? 'top' : undefined }
            definition
            size="large"
          >
            <Table.Body>
              <Table.Row>
                <Table.Cell collapsing>Account Name</Table.Cell>
                <Table.Cell>{accountName}</Table.Cell>
              </Table.Row>
              <Table.Row>
                <Table.Cell collapsing>Network</Table.Cell>
                <Table.Cell>{blockchain.name}</Table.Cell>
              </Table.Row>
              {(ledgerMethod)
                ? (
                  <Table.Row>
                    <Table.Cell>Device Setup</Table.Cell>
                    <Table.Cell>
                      <Segment>
                        {(ledgerMethod === 'all')
                          ? <AccountSetupElementsLedgerOptionsLedgerAll />
                          : false
                        }
                        {(ledgerMethod === 'recover')
                          ? <AccountSetupElementsLedgerOptionsLedgerRecover />
                          : false
                        }
                        {(ledgerMethod === 'use')
                          ? <AccountSetupElementsLedgerOptionsLedgerUse />
                          : false
                        }
                      </Segment>
                      <Button
                        content="Change Ledger Usage"
                        fluid
                        onClick={this.onLedgerReset}
                        size="small"
                      />
                    </Table.Cell>
                  </Table.Row>
                )
                : false
              }
            </Table.Body>
          </Table>
          {(!ledgerMethod)
            ? (
              <Message attached="bottom" icon>
                <Icon name="usb" />
                <Message.Content>
                  <Message.Header>Hardware Wallet Options</Message.Header>
                  <Form.Field style={{ marginTop: '0.5em' }}>
                    <Checkbox
                      checked={!!(ledgerMethod || ledgerSetup)}
                      label="Use a Ledger hardware wallet for this new account?"
                      key={`ledger-${ledgerSetup}-${ledgerMethod}`}
                      name="ledgerSetup"
                      onChange={this.onLedgerSetup}
                      value={!!(ledgerMethod || ledgerSetup)}
                    />
                  </Form.Field>
                </Message.Content>
              </Message>
            )
            : false
          }
          <Form.Field>
            <GlobalButtonElevate
              onSuccess={(password) => onClick(password, ledgerMethod)}
              trigger={(
                <Button
                  content="Create"
                  disabled={disable}
                  fluid
                  primary
                  size="large"
                />
              )}
            />
          </Form.Field>
          <Form.Field>
            <Button
              content="Change"
              fluid
              onClick={onBack}
            />
          </Form.Field>
        </Form>
      </Segment>
    );
  }
}

function mapStateToProps(state) {
  return {
    ledger: state.ledger,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      changeModule,
      ...HardwareLedgerActions
    }, dispatch)
  };
}

export default compose(
  withRouter,
  withTranslation('global'),
  connect(mapStateToProps, mapDispatchToProps)
)(AccountSetupElementsConfirm);
