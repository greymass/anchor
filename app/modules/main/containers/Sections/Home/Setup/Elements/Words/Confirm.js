// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { withTranslation } from 'react-i18next';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import compose from 'lodash/fp/compose';
import { pullAll, sampleSize, shuffle } from 'lodash';

import { Card, Label, Header, Segment } from 'semantic-ui-react';

import { wordlist } from './Wordlist';

class AccountSetupElementsWordsConfirm extends Component<Props> {
  constructor(props) {
    super(props);
    const { words } = props;
    const reducedList = [...wordlist];
    pullAll(reducedList, words);
    const random = sampleSize(reducedList, 6);
    const mixed = shuffle([
      ...words,
      ...random,
    ]);
    this.state = {
      index: 0,
      mixed,
      selected: [],
      words: this.props.words,
    };
  }
  check = (index, word) => {
    const { words } = this.state;
    if (words[index] === word) {
      this.setState({
        index: this.state.index + 1,
        selected: [...this.state.selected, word],
      }, () => {
        if (this.state.selected.length === 6) {
          this.props.onVerified();
        }
      });
    }
  }
  render() {
    const { index, mixed, selected } = this.state;
    return (
      <React.Fragment>
        <Header size="large">
          Verify your encryption key
        </Header>
        <p>Click to select your encryption keywords in order to make sure you wrote them down correctly.</p>
        <Segment secondary>
          <Header>
            {(index >= 6)
              ? (
                <span>Word verified!</span>
              )
              : (
                <span>Select word #{index + 1}...</span>
              )
            }
          </Header>
          <Card.Group itemsPerRow={3}>
            {mixed.map((word, i) => {
              const isSelected = selected.includes(word);
              const selectedIndex = selected.indexOf(word) + 1;
              return (
                <Card key={word}>
                  <Card.Content
                    style={{
                      cursor: 'pointer',
                      padding: 0
                    }}
                    onClick={() => this.check(index, word)}
                  >
                    <Label
                      color={isSelected ? 'blue' : 'grey'}
                      size="large"
                      style={{
                          marginRight: '1em',
                          lineHeight: '1.4285em',
                      }}
                    >
                      {(isSelected)
                        ? (
                          <span>{selectedIndex}</span>
                        )
                        : (
                          <span>&nbsp;</span>
                        )
                      }
                    </Label>
                    <div
                      style={{
                        display: 'inline-block',
                        fontSize: '1.25em',
                        fontWeight: 'bold',
                      }}
                    >
                      {word}
                    </div>
                  </Card.Content>
                </Card>
            );
            })}
          </Card.Group>
        </Segment>
      </React.Fragment>
    );
  }
}

function mapStateToProps(state) {
  return {

  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({

    }, dispatch)
  };
}

export default compose(
  withRouter,
  withTranslation('global'),
  connect(mapStateToProps, mapDispatchToProps)
)(AccountSetupElementsWordsConfirm);
