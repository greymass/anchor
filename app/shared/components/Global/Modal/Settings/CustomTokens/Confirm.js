// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Button, Container, Table } from 'semantic-ui-react';

class GlobalModalSettingsCustomTokensConfirm extends Component<Props> {
  render() {
    const {
      onSubmit,
      t,
      token
    } = this.props;
    return (
      <React.Fragment>
        <Table definition>
          <Table.Row>
            <Table.Cell width={4}>
              {t('global_modal_settings_customtoken_confirm_issuer')}
            </Table.Cell>
            <Table.Cell>
              {token.issuer}
            </Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              {t('global_modal_settings_customtoken_confirm_supply')}
            </Table.Cell>
            <Table.Cell>
              {token.supply}
            </Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              {t('global_modal_settings_customtoken_confirm_maxsupply')}
            </Table.Cell>
            <Table.Cell>
              {token.max_supply}
            </Table.Cell>
          </Table.Row>
        </Table>
        <Container textAlign="center">
          <Button
            color="green"
            content={t('global_modal_settings_customtoken_confirm')}
            onClick={onSubmit}
          />
        </Container>
      </React.Fragment>
    );
  }
}

export default translate('global')(GlobalModalSettingsCustomTokensConfirm);
