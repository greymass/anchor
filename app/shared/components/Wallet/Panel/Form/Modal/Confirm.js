// @flow
import React, { Component } from 'react';
import { I18n } from 'react-i18next';
import { Button, Form, Header, Icon, Input, Modal } from 'semantic-ui-react';

export default class WalletPanelFormModalConfirm extends Component<Props> {
  state = {
    password: ''
  }

  compare = (e, { value }) => this.setState({
    matches: (value === this.props.password),
    password: value
  })

  onKeyPress = (e) => {
    const { matches } = this.state;
    if (matches && e.key === 'Enter') {
      this.props.onSubmit();
    }
  }

  render() {
    const {
      disabled,
      open,
      onCancel,
      onConfirm,
      onSubmit,
      password
    } = this.props;
    const matches = (this.state.password === password);
    return (
      <I18n ns="wallet">
        {
          (t) => (
            <Modal
              centered={false}
              trigger={(
                <Button
                  color="blue"
                  content={t('wallet_panel_password_confirm_button')}
                  disabled={disabled}
                  fluid
                  onClick={onConfirm}
                />
              )}
              onClose={onCancel}
              open={open}
              size="tiny"
            >
              <Header icon="lock" content={t('wallet_panel_password_confirm_title')} />
              <Modal.Content>
                <h3>{t('wallet_panel_password_confirm_description')}</h3>
                <Form.Field
                  autoFocus
                  control={Input}
                  fluid
                  icon={matches ? 'checkmark' : 'x'}
                  label={t('wallet_panel_password_label')}
                  onChange={this.compare}
                  onKeyPress={this.onKeyPress}
                  type="password"
                />
              </Modal.Content>
              <Modal.Actions>
                <Button
                  onClick={onCancel}
                >
                  <Icon name="x" /> {t('cancel')}
                </Button>
                <Button
                  color="green"
                  disabled={!matches}
                  onClick={onSubmit}
                >
                  <Icon name="checkmark" /> {t('wallet_panel_wallet_save')}
                </Button>
              </Modal.Actions>
            </Modal>

          )
        }
      </I18n>
    );
  }
}
