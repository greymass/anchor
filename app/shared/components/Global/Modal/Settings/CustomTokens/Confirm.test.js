import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalModalSettingsCustomTokensConfirm } from './Confirm';

const mockProps = {
  t: jest.fn,
  onSubmit: jest.fn,
  token: { issuer: 'teamgreymass' }
};

describe('GlobalModalSettingsCustomTokensConfirm', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalModalSettingsCustomTokensConfirm {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
