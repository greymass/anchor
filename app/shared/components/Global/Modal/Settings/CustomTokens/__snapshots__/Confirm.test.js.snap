// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalModalSettingsCustomTokensConfirm renders correctly 1`] = `
<Fragment>
  <Table
    as="table"
    definition={true}
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableCell
        as="td"
        width={4}
      >
        <p>
          <Component />
        </p>
      </TableCell>
      <TableCell
        as="td"
      >
        <p>
          teamgreymass
        </p>
      </TableCell>
    </TableRow>
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableCell
        as="td"
      >
        <p>
          <Component />
        </p>
      </TableCell>
      <TableCell
        as="td"
      >
        <p />
      </TableCell>
    </TableRow>
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableCell
        as="td"
      >
        <p>
          <Component />
        </p>
      </TableCell>
      <TableCell
        as="td"
      >
        <p />
      </TableCell>
    </TableRow>
  </Table>
  <Container
    textAlign="center"
  >
    <Button
      as="button"
      color="green"
      content={
        <p>
          [MockFunction]
        </p>
      }
      onClick={[Function]}
    />
  </Container>
</Fragment>
`;
